// @cartesi/rollups/dist/src/types/factories/contracts/history/IHistory__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IHistory__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "_dapp",
                type: "address",
            },
            {
                internalType: "bytes",
                name: "_proofContext",
                type: "bytes",
            },
        ],
        name: "getClaim",
        outputs: [
            {
                internalType: "bytes32",
                name: "epochHash_",
                type: "bytes32",
            },
            {
                internalType: "uint256",
                name: "firstInputIndex_",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "lastInputIndex_",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_consensus",
                type: "address",
            },
        ],
        name: "migrateToConsensus",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes",
                name: "_claimData",
                type: "bytes",
            },
        ],
        name: "submitClaim",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class IHistory__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.IHistory__factory = IHistory__factory;
IHistory__factory.abi = _abi;
