// @cartesi/rollups/dist/src/types/factories/contracts/library/LibOutputValidation__factory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LibOutputValidation__factory = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [],
        name: "IncorrectEpochHash",
        type: "error",
    },
    {
        inputs: [],
        name: "IncorrectOutputHashesRootHash",
        type: "error",
    },
    {
        inputs: [],
        name: "IncorrectOutputsEpochRootHash",
        type: "error",
    },
    {
        inputs: [],
        name: "InputIndexOutOfClaimBounds",
        type: "error",
    },
];
const _bytecode = "0x60566037600b82828239805160001a607314602a57634e487b7160e01b600052600060045260246000fd5b30600052607381538281f3fe73000000000000000000000000000000000000000030146080604052600080fdfea264697066735822122012d1df38d85394b26ba2b48622702197cdc35be000f979ae3bd6440aa6c73afd64736f6c63430008130033";
const isSuperArgs = (xs) => xs.length > 1;
class LibOutputValidation__factory extends ethers_1.ContractFactory {
    constructor(...args) {
        if (isSuperArgs(args)) {
            super(...args);
        }
        else {
            super(_abi, _bytecode, args[0]);
        }
    }
    deploy(overrides) {
        return super.deploy(overrides || {});
    }
    getDeployTransaction(overrides) {
        return super.getDeployTransaction(overrides || {});
    }
    attach(address) {
        return super.attach(address);
    }
    connect(signer) {
        return super.connect(signer);
    }
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.LibOutputValidation__factory = LibOutputValidation__factory;
LibOutputValidation__factory.bytecode = _bytecode;
LibOutputValidation__factory.abi = _abi;
