// @cartesi/rollups/dist/src/types/factories/contracts/relays/IDAppAddressRelay__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IDAppAddressRelay__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [],
        name: "getInputBox",
        outputs: [
            {
                internalType: "contract IInputBox",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_dapp",
                type: "address",
            },
        ],
        name: "relayDAppAddress",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class IDAppAddressRelay__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.IDAppAddressRelay__factory = IDAppAddressRelay__factory;
IDAppAddressRelay__factory.abi = _abi;
