// @cartesi/rollups/dist/src/types/factories/contracts/portals/IERC721Portal__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IERC721Portal__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "contract IERC721",
                name: "_token",
                type: "address",
            },
            {
                internalType: "address",
                name: "_dapp",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_tokenId",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "_baseLayerData",
                type: "bytes",
            },
            {
                internalType: "bytes",
                name: "_execLayerData",
                type: "bytes",
            },
        ],
        name: "depositERC721Token",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [],
        name: "getInputBox",
        outputs: [
            {
                internalType: "contract IInputBox",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class IERC721Portal__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.IERC721Portal__factory = IERC721Portal__factory;
IERC721Portal__factory.abi = _abi;
