// @cartesi/rollups/dist/src/types/factories/contracts/portals/IERC1155BatchPortal__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IERC1155BatchPortal__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "contract IERC1155",
                name: "_token",
                type: "address",
            },
            {
                internalType: "address",
                name: "_dapp",
                type: "address",
            },
            {
                internalType: "uint256[]",
                name: "_tokenIds",
                type: "uint256[]",
            },
            {
                internalType: "uint256[]",
                name: "_values",
                type: "uint256[]",
            },
            {
                internalType: "bytes",
                name: "_baseLayerData",
                type: "bytes",
            },
            {
                internalType: "bytes",
                name: "_execLayerData",
                type: "bytes",
            },
        ],
        name: "depositBatchERC1155Token",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [],
        name: "getInputBox",
        outputs: [
            {
                internalType: "contract IInputBox",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class IERC1155BatchPortal__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.IERC1155BatchPortal__factory = IERC1155BatchPortal__factory;
IERC1155BatchPortal__factory.abi = _abi;
