// @cartesi/rollups/dist/src/types/factories/contracts/dapp/ICartesiDAppFactory__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ICartesiDAppFactory__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "contract IConsensus",
                name: "consensus",
                type: "address",
            },
            {
                indexed: false,
                internalType: "address",
                name: "dappOwner",
                type: "address",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "templateHash",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "contract CartesiDApp",
                name: "application",
                type: "address",
            },
        ],
        name: "ApplicationCreated",
        type: "event",
    },
    {
        inputs: [
            {
                internalType: "contract IConsensus",
                name: "_consensus",
                type: "address",
            },
            {
                internalType: "address",
                name: "_dappOwner",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "_templateHash",
                type: "bytes32",
            },
            {
                internalType: "bytes32",
                name: "_salt",
                type: "bytes32",
            },
        ],
        name: "calculateApplicationAddress",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract IConsensus",
                name: "_consensus",
                type: "address",
            },
            {
                internalType: "address",
                name: "_dappOwner",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "_templateHash",
                type: "bytes32",
            },
            {
                internalType: "bytes32",
                name: "_salt",
                type: "bytes32",
            },
        ],
        name: "newApplication",
        outputs: [
            {
                internalType: "contract CartesiDApp",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract IConsensus",
                name: "_consensus",
                type: "address",
            },
            {
                internalType: "address",
                name: "_dappOwner",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "_templateHash",
                type: "bytes32",
            },
        ],
        name: "newApplication",
        outputs: [
            {
                internalType: "contract CartesiDApp",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class ICartesiDAppFactory__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.ICartesiDAppFactory__factory = ICartesiDAppFactory__factory;
ICartesiDAppFactory__factory.abi = _abi;
