// @cartesi/rollups/dist/src/types/factories/contracts/consensus/AbstractConsensus__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AbstractConsensus__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "address",
                name: "application",
                type: "address",
            },
        ],
        name: "ApplicationJoined",
        type: "event",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "_dapp",
                type: "address",
            },
            {
                internalType: "bytes",
                name: "_proofContext",
                type: "bytes",
            },
        ],
        name: "getClaim",
        outputs: [
            {
                internalType: "bytes32",
                name: "epochHash_",
                type: "bytes32",
            },
            {
                internalType: "uint256",
                name: "firstInputIndex_",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "lastInputIndex_",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [],
        name: "join",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class AbstractConsensus__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.AbstractConsensus__factory = AbstractConsensus__factory;
AbstractConsensus__factory.abi = _abi;
